### PR TITLE
(8/10) RUM-2895 Create the TextCompositionGroupMapper

### PR DESCRIPTION
**Pull request by xgouchet**
_This PR has been created automatically by splitting a large branch_

RUM-2895 Create the TextCompositionGroupMapper

### CODE REVIEW

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            ),
+            null
+        )
+    }
+
+    /**
+     * Parse Compose's [androidx.compose.ui.unit.TextUnitType] packed value.
+     */
+    protected fun convertTextUnit(dimen: Long): Float? {
+        return if (dimen == 0x7FC0_0000L) {
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt:82]: need to comment this magic value

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        }
+    }
+
+    companion object {
+
+        private const val VALUE_MASK = 0xFFFF_FFFFL
+        private const val UNIT_MASK = 0xFFL shl 32 // 0xFF_0000_0000
+        private const val UNIT_TYPE_UNSPECIFIED = 0x00L shl 32 // 0x00_0000_0000
+        private const val UNIT_TYPE_SP = 0x01L shl 32 // 0x01_0000_0000
+        private const val UNIT_TYPE_EM = 0x02L shl 32 // 0x2_0000_0000
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt:103]: minor: for the consistency with comments format above

```suggestion
        private const val UNIT_TYPE_EM = 0x02L shl 32 // 0x02_0000_0000
```

**Review by 0xnm — DISMISSED** [review]

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    protected fun convertTextUnit(dimen: Long): Float? {
+        return if (dimen == 0x7FC0_0000L) {
+            null
+        } else {
+            val unit = dimen and UNIT_MASK
+            val value = Float.fromBits((dimen and VALUE_MASK).toInt())
+
+            when (unit) {
+                UNIT_TYPE_UNSPECIFIED,
+                UNIT_TYPE_SP -> value
```
Comment (mariusc83) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt:90]: don't we need to normalize it with the density here ?

**Review by jonathanmos — APPROVED** [review]

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt (hunk 3 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt:90]: Eventually yes, that'll be a different task

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt (hunk 1 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TextCompositionGroupMapper.kt:82]: Yes, it's a `NaN`

**Review by 0xnm — APPROVED** [review]

**Pull request merged**